nest-tax-backend/src/tax/utils/unified-tax.util.ts

```diff
@@ -306,46 +288,23 @@ const calculateInstallmentStatus = (
     !dueDateSecondInstallmentInFuture &&
     installmentAmounts[1].status !== InstallmentPaidStatusEnum.PAID &&
     installmentAmounts[1].status !== InstallmentPaidStatusEnum.OVER_PAID
+
   // First installment status logic
-  const firstInstallmentStatus =
-    isFirstInstallmentLate || isSecondInstallmentLate
-      ? InstallmentPaidStatusEnum.AFTER_DUE_DATE
-      : installmentAmounts[0].status
-  const firstInstallmentRemainingAmount =
-    isFirstInstallmentLate || isSecondInstallmentLate
-      ? 0
-      : installmentAmounts[0].toPay
+  const firstInstallmentStatus = isFirstInstallmentLate
+    ? InstallmentPaidStatusEnum.AFTER_DUE_DATE
+    : installmentAmounts[0].status
+  const firstInstallmentRemainingAmount = installmentAmounts[0].toPay
 
   // Second installment status logic
-  let secondInstallmentStatus: InstallmentPaidStatusEnum
-  let secondInstallmentRemainingAmount: number
-
-  if (isSecondInstallmentLate) {
-    secondInstallmentStatus = InstallmentPaidStatusEnum.AFTER_DUE_DATE
-    secondInstallmentRemainingAmount = 0
-  } else if (isFirstInstallmentLate) {
-    secondInstallmentStatus = installmentAmounts[0].status
-    secondInstallmentRemainingAmount =
-      installmentAmounts[1].toPay + installmentAmounts[0].toPay
-  } else {
-    secondInstallmentStatus = installmentAmounts[1].status
-    secondInstallmentRemainingAmount = installmentAmounts[1].toPay
-  }
+  const secondInstallmentStatus = isSecondInstallmentLate
+    ? InstallmentPaidStatusEnum.AFTER_DUE_DATE
+    : installmentAmounts[1].status
+  const secondInstallmentRemainingAmount = installmentAmounts[1].toPay
 
   // Third installment status logic
-  let thirdInstallmentStatus: InstallmentPaidStatusEnum
-  let thirdInstallmentRemainingAmount: number
-
-  if (isSecondInstallmentLate) {
-    thirdInstallmentStatus = installmentAmounts[1].status
-    thirdInstallmentRemainingAmount = installmentAmounts.reduce(
-      (agg, i) => i.toPay + agg,
-      0,
-    )
-  } else {
-    thirdInstallmentStatus = installmentAmounts[2].status
-    thirdInstallmentRemainingAmount = installmentAmounts[2].toPay
-  }
+  const thirdInstallmentStatus = installmentAmounts[2].status
```

Comment:
Just checking,  `firstInstallmentStatus` and `secondInstallmentStatus` can have value of `InstallmentPaidStatusEnum.AFTER_DUE_DATE`. 
`thirdInstallmentStatus` can't have that status, is it ok?